File: src/sources/medicalxpress.py
import feedparser
from .base import BaseSource, RawArticle


class MedicalXpressSource(BaseSource):
    name = "MedicalXpress"
    language = "en"
    FEEDS = [
        "https://medicalxpress.com/rss-feed/",
        "https://medicalxpress.com/rss-feed/medical-research-news/",
        "https://medicalxpress.com/rss-feed/diseases-conditions-news/",
    ]

    async def fetch(self) -> list[RawArticle]:
        articles = []
        for url in self.FEEDS:
            try:
                feed = feedparser.parse(url)
                for entry in feed.entries:
                    articles.append(RawArticle(
                        source=self.name,
                        title=entry.get("title", "").strip(),
                        url=entry.get("link", ""),
                        published_at=self._safe_date(entry.get("published")),
                        summary=entry.get("summary", ""),
                        language=self.language,
                    ))
            except Exception as e:
                from src.utils.logger import logger
                logger.warning(f"[MedicalXpress] 피드 오류 {url}: {e}")
        return articles

File: src/sources/medicalxpress.ts
import Parser from 'rss-parser';
import { BaseSource, RawArticle } from './base';
import { logger } from '@/lib/logger';

const parser = new Parser({ timeout: 15000, maxRedirects: 5 });

const FEEDS = [
  'https://medicalxpress.com/rss-feed/',
  'https://medicalxpress.com/rss-feed/medical-research-news/',
  'https://medicalxpress.com/rss-feed/diseases-conditions-news/',
];

export class MedicalXpressSource extends BaseSource {
  readonly name = 'MedicalXpress';

  async fetch(): Promise<RawArticle[]> {
    const articles: RawArticle[] = [];
    for (const feedUrl of FEEDS) {
      try {
        const feed = await parser.parseURL(feedUrl);
        for (const item of feed.items) {
          if (!item.title || !item.link) continue;
          articles.push({
            source: this.name,
            title: this.stripHtml(item.title),
            url: item.link,
            published_at: this.parseDate(item.pubDate ?? item.isoDate),
            summary: item.contentSnippet ?? '',
            language: this.language,
          });
        }
      } catch (e) {
        logger.warn(`[MedicalXpress] 피드 오류 ${feedUrl}: ${e}`);
      }
    }
    return articles;
  }
}
